core/memory_tracker/cc/memory_tracker.h

```diff
@@ -15,19 +15,26 @@
  */
 
 #include "core/cc/target.h"
+#include "core/memory_tracker/cc/memory_protections.h"
 
 #define COHERENT_TRACKING_ENABLED
-
-#ifndef GAPII_MEMORY_TRACKER_H
-#define GAPII_MEMORY_TRACKER_H
-
+#ifdef COHERENT_TRACKING_ENABLED
```

Comment:
Nit: I prefer to use:

```c++
#define COHERENT_TRACKING_ENABLED 1
#if COHERENT_TRACKING_ENABLED
```

As I believe you can configure the compile to warn if the macro doesn't exist.